File: go_services/process_service_test.go
package go_services_test

import (
	. "github.com/onsi/ginkgo"
	. "github.com/onsi/gomega"
	"github.com/sporto/service_objects/go_services"
)

type FakeFindService struct {
}

func (c *FakeFindService) Run() []int {
	return []int{1, 2, 3}
}

var _ = Describe("ProcessService", func() {
	var (
		serv *go_services.ProcessService
	)

	BeforeEach(func() {
		serv = &go_services.ProcessService{}
	})

	It("uses the defaults", func() {
		res := serv.Run(2)
		Expect(res).To(Equal([]int{2, 4}))
	})

	It("can take a fake", func () {
		// change dependecies
		serv.FindService = new(FakeFindService)

		res := serv.Run(2)
		Expect(res).To(Equal([]int{2, 4, 6}))
	})
})

File: js/process_service_test.js
// var assert = require("assert");
var Service = require('./process_service');
var expect = require('chai').expect;

describe('ProcessService', function(){

	var serv;

	beforeEach(function () {
		serv = Service();
	});

	describe('.run', function(){
		it('uses the defaults', function(){
			var res = serv.run(2);
			expect(res).to.eql([2, 4]);
		});

		it('can take a fake', function () {
			var fakeFindService = {
				run: function () {
					return [1, 2, 3];
				}
			}
			serv.findService = fakeFindService;
			var res = serv.run(2);
			expect(res).to.eql([2, 4, 6]);
		});
	});
})